src/v3/src/components/TextWithLink/TextWithLink.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Box } from '@mui/material';
+import { h } from 'preact';
+import { useEffect } from 'preact/hooks';
+
+import { useOnSubmit } from '../../hooks';
+import {
+  TextWithLinkElement,
+  UISchemaElementComponent,
+} from '../../types';
+
+const TextWithLink: UISchemaElementComponent<{
+  uischema: TextWithLinkElement
+}> = ({ uischema }) => {
+  const {
+    content,
+    step,
+    stepToRender,
+    linkClass,
+    action: actionFn,
+  } = uischema.options;
+  const onSubmitHandler = useOnSubmit();
+
+  useEffect(() => {
+    const handleClick = async (event: Event) => {
+      event.preventDefault();
+      onSubmitHandler({
+        includeData: Boolean(false),
+        step,
+        stepToRender,
+        actionFn,
+      });
+    };
+
+    const [element] = Array.from(document.getElementsByClassName(linkClass));
+    element.addEventListener('click', handleClick);
+
+    return () => {
+      element.removeEventListener?.('click', handleClick);
+    };
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <Box>
+      <Box dangerouslySetInnerHTML={{ __html: content }} />
```

Comment:
Based on this implementation I might be more inclined to call this component `TextWithHtml`

Reply:
That makes perfect sense. 🙏  Thank you! Will update.